File: 05/aoc_05.py

def get_id(ec):

    i = left = 0
    right = 127
    
    while(i < 7):

        mid = left + (right - left)//2
        if ec[i] == 'F':
            right = mid
        else:
            left = mid

        i += 1

    row = right

    i = 7
    left = 0
    right = 7
    while(i < len(ec)):
        
        mid = left + (right - left)//2
        if ec[i] == 'L':
            right =  mid
        else:
            left = mid

        i += 1
    col = right


    return row * 8 + col

max_val = float('-inf')
min_val = float('inf')
su = 0
with open('in.txt','r') as f:
        ln = f.readline().rstrip()
        while(ln):
            _id = get_id(ln)
            max_val = max(max_val, int(_id))
            min_val = min(min_val, int(_id))
            su += int(_id)
            ln = f.readline().rstrip()

print(min_val,max_val)
t = sum(range(8,977))
print(t-su)

File: 05/aoc_05.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

var (
	ln string
)

func getID(ec string) int {

	var l = 0
	var r = 127
	for i := 0; i < 7; i++ {

		mid := l + (r-l)/2
		if ec[i] == 'F' {
			r = mid
		} else {
			l = mid
		}
	}
	row := r

	l = 0
	r = 7
	for i := 7; i < len(ec); i++ {
		mid := l + (r-l)/2
		if ec[i] == 'L' {
			r = mid
		} else {
			l = mid
		}
	}
	col := r

	return row*8 + col
}

func main() {

	// file open & close
	f, _ := os.Open("./in.txt")
	defer func() {
		f.Close()
	}()

	// problem specific vars
	maxVal := 0
	minVal := math.MaxInt64
	sum := 0.0

	// going through the lines
	s := bufio.NewScanner(f)
	for s.Scan() {
		ln = s.Text()

		id := float64(getId(ln))
		maxVal = int(math.Max(float64(maxVal), id))
		minVal = int(math.Min(float64(minVal), id))

		sum += id
	}

	fmt.Printf("Ans part 1: %d\n", maxVal)

	t := 0
	for n := minVal; n <= maxVal; n++ {
		t += n
	}

	fmt.Printf("Ans part 2: %d\n", t-int(sum))
}
